site/app/libraries/database/DatabaseQueries.php

```diff
@@ -6159,4 +6159,30 @@ function ($a) {
         }
         return '';
     }
+
+    /**
+     * Delete user from DBs identified by user_id, semester, and course.
+     *
+     * Query issues DELETE on master DB's courses_users table.  When potentially
+     * successful, trigger function will attempt to delete user from course DB.
+     * If the trigger fails, it is expected that user is also not removed from
+     * master DB.  When user cannot be deleted (probably due to referential
+     * integrity), query is expected to return 0 rows to indicate user was not
+     * deleted.
+     *
+     * @param string $user_id
+     * @param string $semester
+     * @param string $course
+     * @return bool false on failure (or 0 rows deleted), true otherwise.
+     */
+    public function deleteUser($user_id, $semester, $course) {
+        // User cannot remove themself.
+        if ($user_id === $this->core->getUser()->getId()) {
+            return false;
+        }
```

Comment:
I think we should move this to the controller itself and then display an appropriate error message. It's weird that if I go to delete myself, I'll always see "Could not remove {$displayed_fullname}.  They may have recorded activity in your course." as that's not technically the error.